rwengine/src/render/GameRenderer.hpp

```diff
@@ -140,8 +140,9 @@ class GameRenderer {
     void setupRender();
     void renderPostProcess();
 
-    Renderer* getRenderer() {
-        return renderer.get();
+    Renderer*& getRenderer() {
```

Comment:
This pull request aims to prevent memory leak. 
Practically in the code are only pointers, so this refactor should implicate to change them.
And this is a lot of work to do, to much for this pull.